source/localizable/tutorial/service.md

```diff
@@ -0,0 +1,289 @@
+Lets start by appending the component as we want it to the end of the `rental-listing` template, and provide it with the city of the current rental.
+
+```app/templates/components/rental-listing.hbs
+<h2>{{rental.title}}</h2>
+<p>Owner: {{rental.owner}}</p>
+<p>Type: {{rental-property-type rental.type}} - {{rental.type}}</p>
+<p>Location: {{rental.city}}</p>
+<p>Number of bedrooms: {{rental.bedrooms}}</p>
+{{#if isImageShowing}}
+  <p>
+    <img alt={{rental.type}} src={{rental.image}} width="500"/>
+  </p>
+  <button {{action "imageHide"}}>Hide image</button>
+{{else}}
+  <button {{action "imageShow"}}>Show image</button>
+{{/if}}
+{{location-map location=rental.city}}
+```
+
+Next step is to get our page compiling by using Ember CLI to generate this component.
+
+```shell
+ember g component location-map
+```
+
+The Ember CLI component generator will create for us a test file, a component JavaScript file, and a template.
+To think about what we want our component to do, will write a testcase.
+In this case we want our Google Maps service to provide us with the map display, so all we want our component to do is take the output of our maps service, which is a map element, and append it to an element in the component's template.
+
+All we want to do in our component is make sure its appending the output of the maps service to an inner map container element.
+To limit the test to validating just this behavior we'll take advantage of the registration api to provide a mock amps service.
+In the mock service, define a method that will fetch the map based on location, called `getMapElement`.
+
+```tests/integration/components/location-map-test.js
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import Ember from 'ember';
+
+let MockMapsService = Ember.Service.extend({
+  getMapElement(location) {
+    this.set('calledWithLocation', location);
+    return document.createElement('div');
+  }
+});
+
+moduleForComponent('location-map', 'Integration | Component | location map', {
+  integration: true,
+  beforeEach() {
+    this.register('service:maps', MockMapsService);
+    this.inject.service('maps', { as: 'mapsService' });
+  }
+});
+
+test('should append map element to container element', function(assert) {
+  this.set('myLocation', 'New York');
+  this.render(hbs`{{location-map location=myLocation}}`);
+  assert.equal(this.$('.map-container').children().length, 1, 'container should have one child');
+  assert.equal(this.get('mapsService.calledWithLocation'), 'New York', 'should call service with New York');
+});
+```
+
+Notice that our mock service above sets a variable called `calledWithLocation`.  We later assert on this variable to ensure that the service is being called with the same location provided to the component.
+
+To get the test to pass, add the container element to the component template.
+
+```app/templates/components/location-map.hbs
+<div class="map-container"/>
```

Comment:
I don't think self-closing `<div>` tags are valid. As I understand it, only [void elements](https://www.w3.org/TR/html-markup/syntax.html#void-element) can omit the end-tag.